src/first/Main.java

```diff
@@ -15,11 +15,17 @@
     public static void main(String[] args) throws Exception {
         listf(PATH, files);
         threads = new FastFileSearch[NUM_OF_THREADS];
-        for (int i = 0; i < NUM_OF_THREADS; i++) {
-            threads[i] = new FastFileSearch("Thread #" + i, files.get(i), phrase);
-        }
-        for (int i = 0; i < NUM_OF_THREADS; i++) {
-            threads[i].thrd.join();
+        FastFileSearch.codePhrase = phrase;
+        while (FastFileSearch.counter != files.size()) {
+            int i;
+            int border = (files.size() - FastFileSearch.counter) % NUM_OF_THREADS == 0 ?
+                    NUM_OF_THREADS : (files.size() - FastFileSearch.counter) % NUM_OF_THREADS;
+            for (i = 0; i < border; i++) {
+                threads[i] = new FastFileSearch("Thread #" + i, files.get(FastFileSearch.counter), phrase);
+                threads[i].thrd.join();
```

Comment:
its a not multithreading. You running thread and waiting when he done. You must run all threads